gammapy/spectrum/utils.py

```diff
@@ -94,63 +94,36 @@ def bin_edges(self, energy):
         return pix1, pix2, energy1, energy2
 
 
-# Todo: find a better place for these functions (Spectrum analysis class?)
-def plot_exclusion_mask(**kwargs):
-    """Plot exclusion mask
+def calculate_predicted_counts(model, aeff, edisp, livetime):
+    """Get npred 
 
-    The plot will be centered at the pointing position
+    The true and reco energy binning are inferred from the provided IRFs.
 
-    Parameters
-    ----------
-    size : `~astropy.coordinates.Angle`
-    Edge length of the plot
-    """
-    from gammapy.image import ExclusionMask
-    from gammapy.spectrum import SpectrumExtraction
-    # Todo: plot exclusion mask as contours with skymap class
-
-    exclusion = ExclusionMask.from_fits(SpectrumExtraction.EXCLUDEDREGIONS_FILE)
-    ax = exclusion.plot(**kwargs)
-    return ax
-
-
-def plot_npred_vs_excess(ogip_dir='ogip_data', npred_dir='n_pred', ax=None):
-    """Plot predicted and measured excess counts
+    TODO: make energy binning optional once `~gammapy.irf.EnergyDispersion`
+    has an evaluate method.
 
     Parameters
     ----------
-    npred_dir : str, Path
-        Directory holding npred fits files
-    ogip_dir : str, Path
-        Directory holding OGIP data
+    model : `~gammapy.spectrum.models.SpectralModel`
+        Spectral model
+    livetime : `~astropy.units.Quantity`
+        Observation duration
+    aeff : `~gammapy.irf.EffectiveAreaTable`
+        EffectiveArea
+    edisp : `~gammapy.irf.EnergyDispersion`, optional
+        EnergyDispersion
     """
-    from ..spectrum.spectrum_extraction import SpectrumObservationList
-    from ..spectrum import CountsSpectrum
-    from ..utils.scripts import make_path
-
-    import matplotlib.pyplot as plt
-    ax = plt.gca() if ax is None else ax
-
-    ogip_dir = make_path(ogip_dir)
-    n_pred_dir = make_path(npred_dir)
-
-    obs = SpectrumObservationList.read_ogip(ogip_dir)
-    excess = np.sum([o.excess_vector for o in obs])
-
-    # Need to give RMF file for reco energy binning
-    id = obs[0].meta.obs_id
-    rmf = str(ogip_dir / 'rmf_run{}.fits'.format(id))
-    val = [CountsSpectrum.read_bkg(_, rmf) for _ in n_pred_dir.glob('*.fits')]
-    npred = np.sum(val)
-
-    npred.plot(ax=ax, color='red', alpha=0.7, label='Predicted counts')
-    excess.plot(ax=ax, color='green', alpha=0.7, label='Excess counts')
-    ax.legend(numpoints=1)
-    plt.xscale('log')
-
-    return ax
+    from . import CountsSpectrum
 
+    true_energy = aeff.energy.data
+    flux = model.integral(emin=true_energy[:-1], emax=true_energy[1:]) 
+    counts = flux * livetime * aeff.evaluate()
+    counts = counts.decompose()
```

Comment:
I find this `decompose` stuff a bit weird and complicated to look at.
Maybe make it an array via `counts = counts.to('').value`?
(Not sure this is better)